src/dialectic/core.py

```
@@ -193,87 +96,30 @@ async def _prefetch_relevant_observations(self, query: str) -> str | None:
             if not explicit_repr.is_empty():
                 parts.append(explicit_repr.format_as_markdown(include_ids=False))
-
             if not derived_repr.is_empty():
-                # Include IDs for derived so agent can use get_reasoning_chain
                 parts.append(derived_repr.format_as_markdown(include_ids=True))
 
             return "\n".join(parts)
 
+        except ValidationException:
+            raise
         except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)